File: programacion 3/Leccion02/BloquesInicializacion/src/domain/Persona.java
package domain;
public class Persona {
    private final int idPersona;
    private static int contadorPersona;
    
    static {    //Bloque de inicialización estático
        System.out.println("Ejecución bloque estático");
        ++Persona.contadorPersona;  //no es necesario poner el nombre de la clase Persona
        //idPersona = 10;   NO es un atributo estático, por eso da error. No se puede usar this
    }
    {//Bloque de inicialización NO estático( o contexto dinamico)
        System.out.println("Ejercución del bloque NO estático");
        this.idPersona = Persona.contadorPersona++; //Incrementamos el atributo
    }
    //Los bloques de inicialización se ejecutan antes del constructor
    public Persona(){   //constructor
        System.out.println("Esta es la ejecución del constructor");
    }
    public int getIdPersona (){
        return this.idPersona;
    }

    @Override
    public String toString() {
        return "Persona{" + "idPersona=" + idPersona + '}';
    }
    
}

File: Laboratorio 3/JavaScript/Leccion07/Persona.js
class Persona{ 

    static contadorPersonas = 0;

    constructor(nombre, apellido, edad){
        this._idPersona = ++Persona.contadorPersonas;
        this._nombre = nombre;
        this._apellido = apellido;
        this._edad = edad;
    }
    get idPersona(){
        return this._idPersona;
    }

    get nombre(){
        this._nombre;
    }
    
    set nombre(nombre){
        this._nombre = this.nombre;
    }

    get apellido(){
        this._apellido;
    }
    
    set apellido(apellido){
        this._apellido = this.apellido;
    }

    get edad(){
        this._edad;
    }
    
    set edad(edad){
        this._edad = this.edad;
    }

    toString(){
        // return this._idPersona+' '+this_nombre+' '+this._apellido+' '+this._edad; O TAMBIEN
        return `
        ${this._idPersona}
        ${this._nombre}
        ${this._apellido}
        ${this._edad}`;
    }

}